### PR TITLE
czkawka: add .desktop and icons

### LINKED ISSUE #262665: Czkawka missing .desktop file for GUI
### Describe the bug
When Czkawka is installed via home-manager, no .desktop files are made for the GUI part of the application, meaning you have to run `czkawka_gui` from the terminal to run the application.

### Steps To Reproduce
1. Install Czkawka, i.e. 
```nix
home.packages = [
  pkgs.czkawka
]
```
2. Search for a .desktop file for Czkawka, `sudo find / -name "*.desktop" | grep "czk"`

### Expected behavior
A desktop file should be there so you can open the Czkawka GUI from things like an application launcher.

### Additional context
Along with this request, I'm providing three files from the czkawka-git package I got from the AUR in the hopes that it might help:
1. The .desktop file they provided, with .txt appended to the filetype to share on Github 
2. The 128x128 png icon they provided, renamed with 128x128 appended to the name
3. The 64x64 png icon they provided, renamed with 64x64 appended to the name 

I'm not sure if these would get used in nixpkgs, but figured it wouldn't hurt to provide them.

### Notify maintainers

@yanganto @0x4A6F 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.57, NixOS, 23.05 (Stoat), 23.05.4335.898cb2064b6e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.5`
 - channels(kody): `"home-manager-23.05.tar.gz"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```
[com.github.qarmin.czkawka.desktop.txt](https://github.com/NixOS/nixpkgs/files/13062298/com.github.qarmin.czkawka.desktop.txt)
com github qarmin czkawka128x128
com github qarmin czkawka64x64

### PR DESCRIPTION
**Pull request by yanganto**
## Description of changes
Fix #262665

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chewblacka** [comment]
LGTM 
Package builds and desktop file gets installed with correct icon.

```
1 package built:
czkawka
```

**Review by chewblacka — APPROVED** [review]
LGTM. Package builds with working desktop file and icon.

**Review by khaneliman — APPROVED** [review]

**Comment by yanganto** [comment]
@chewblacka thanks for the review.

File: pkgs/tools/misc/czkawka/default.nix [hunk 1]
```diff
@@ -65,6 +65,18 @@ rustPlatform.buildRustPackage rec {
     command = "czkawka_cli --version";
   };
 
+  postInstall = ''
+    # Install Icons
+    install -D data/icons/com.github.qarmin.czkawka.svg \
+      $out/share/icons/hicolor/scalable/apps/com.github.qarmin.czkawka.svg
+    install -D data/icons/com.github.qarmin.czkawka-symbolic.svg \
+      $out/share/icons/hicolor/scalable/apps/com.github.qarmin.czkawka-symbolic.svg
```
Comment (0x4A6F) [inline pkgs/tools/misc/czkawka/default.nix:73]: This file isn't used AFAIK.

There is also a metainfo we can install:
```
install -Dm444 data/com.github.qarmin.czkawka.metainfo.xml -t $out/share/metainfo
```

File: pkgs/tools/misc/czkawka/default.nix [hunk 2]
```diff
@@ -65,6 +65,18 @@ rustPlatform.buildRustPackage rec {
     command = "czkawka_cli --version";
   };
 
+  postInstall = ''
+    # Install Icons
+    install -D data/icons/com.github.qarmin.czkawka.svg \
+      $out/share/icons/hicolor/scalable/apps/com.github.qarmin.czkawka.svg
```
Comment (0x4A6F) [inline pkgs/tools/misc/czkawka/default.nix:71]:
```diff
-    install -D data/icons/com.github.qarmin.czkawka.svg \
-     $out/share/icons/hicolor/scalable/apps/com.github.qarmin.czkawka.svg
+    install -Dm444 -t $out/share/icons/hicolor/scalable/apps data/icons/com.github.qarmin.czkawka.svg
```

File: pkgs/tools/misc/czkawka/default.nix [hunk 3]
```diff
@@ -65,6 +65,18 @@ rustPlatform.buildRustPackage rec {
...
+    install -D data/icons/com.github.qarmin.czkawka-symbolic.svg \
+      $out/share/icons/hicolor/scalable/apps/com.github.qarmin.czkawka-symbolic.svg
+
+    # Install Desktop Entry
+    install -D data/com.github.qarmin.czkawka.desktop \
+      $out/share/applications/com.github.qarmin.czkawka.desktop
```
Comment (0x4A6F) [inline pkgs/tools/misc/czkawka/default.nix:77]: We could also use `makeDesktopItem`, but I see no real mentioning in `doc`.

```diff
-    install -D data/com.github.qarmin.czkawka.desktop \
-      $out/share/applications/com.github.qarmin.czkawka.desktop
+    install -Dm444 -t $out/share/applications data/com.github.qarmin.czkawka.desktop
```

**Review by 0x4A6F — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/czkawka/default.nix (hunk 3 above)
Comment (h7x4) [reply pkgs/tools/misc/czkawka/default.nix:77]: I think it's usually better to use upstream dist materials when available, and rather override any parts that won't fit. The less we need to maintain, the better.

**Review by eliandoran — APPROVED** [review]

**Review by 0x4A6F — APPROVED** [review]
Thanks

**Pull request merged**